### ISSUE TITLE
No use on Android

### ISSUE THREAD

**Issue by chris109b** [espeak-ng/espeak-ng#1078]
Environment:
CalyxOS, Android 12, German, Installation from F-droid.

There are 0 languages/voices installed. There is no feature to download any.

There is a button, to install voices from sd card, but no instructions to get the files on to the sd-card in the first place.

After stealing the data directory of espeak-ng from a Linux PC, it turns out, the install from sd-card button doesn't do anything.

**Comment by alex19EP** [comment]
hello, Let me try to explain what the problem is.
to successfully compile espeak-ng for android, you need to create an archive with compiled data files. autoconf and make are currently used for this. i have plans to fix this situation see #960

to fix fdroid now you need to tell them that they need to use native espeak to compile data files then make to create data archive and only then build android app with gradle.

**Comment by archtur** [comment]
> hello, Let me try to explain what the problem is. to successfully compile espeak-ng for android, you need to create an archive with compiled data files. autoconf and make are currently used for this. i have plans to fix this situation see #960
> 
> to fix fdroid now you need to tell them that they need to use native espeak to compile data files then make to create data archive and only then build android app with gradle.

Hey @alex19EP,

is this the same root cause as for issue #974?

**Comment by alex19EP** [comment]
> > hello, Let me try to explain what the problem is. to successfully compile espeak-ng for android, you need to create an archive with compiled data files. autoconf and make are currently used for this. i have plans to fix this situation see #960
> > to fix fdroid now you need to tell them that they need to use native espeak to compile data files then make to create data archive and only then build android app with gradle.
> 
> Hey @alex19EP,
> 
> is this the same root cause as for issue #974?

yes.

**Comment by alex19EP** [comment]
> hello, Let me try to explain what the problem is. to successfully compile espeak-ng for android, you need to create an archive with compiled data files. autoconf and make are currently used for this. i have plans to fix this situation see #960
> 
> to fix fdroid now you need to tell them that they need to use native espeak to compile data files then make to create data archive and only then build android app with gradle.

I looked at espeak in fdroid. I was wrong when I said that they are not building the application correctly. however, for some reason espeak doesn't really get initialized the first time it starts up. Perhaps updating to a newer version will fix this.

**Comment by ominfowave** [comment]
Hi @chris109b , any luck with this? I am also facing the same issue, I have already compiled espeak-ng using VS and the build is successful, but still, there are no voice data, on clicking on Speak button it just used Androids default TTS to speak, I don't see any use of SpeechSynthesis here!

**Comment by chris109b** [comment]
Nope, I don't have an development environment for Android and I don't have the resources to widen my scope in this direction.

**Comment by IzzySoft** [comment]
> Perhaps updating to a newer version will fix this.

Good hint @alex19EP – but see #1185 … Until that is solved, end users have no chance to follow that advice. Oh, except when downloading the APK from `releases/` – but for existing F-Droid users, the signature won't match, so they'll need to uninstall and reinstall (and reconfigure I guess). F-Droid still has the 1.51-dev from 4/2021 as when making the 1.51 release in 4/2022, `versionCode` was not increased. And while `build.gradle` allready says `1.52-dev`, `versionCode` [still is left on `20`](https://github.com/espeak-ng/espeak-ng/blob/master/android/build.gradle#L73). Maybe `versionCode` could be increased and a new tag created?

**Comment by valerio-bozzolan** [comment]
Just to say that I can reproduce the problem on a fresh installation of 1.51 from F-Droid.

@chris109b i suggest to rename this to "Android: 0 languages installed (not working)" or something like that a bit more descriptive.

**Comment by akostadinov** [comment]
Is anybody looking at this?

**Comment by vanMacG** [comment]
@akostadinov
I updated to 1.51.1 from F-Droid and my explanation from [#974](https://github.com/espeak-ng/espeak-ng/issues/974#issuecomment-1081106807) still works fine.

**Comment by animeshahilya** [comment]
@alex19EP can i get your build? wanted to serculate in to the network of blind people. they will get benifited with your optimizations.

**Comment by qkqsb** [comment]
@alex19EP I can confirm your solution worked for me. Thank you for the help. Are there updates you have as to when your merge will happen that fixes this in a more complete way?

**Comment by jianboy** [comment]
**there is no voice installed.**

![image](https://github.com/espeak-ng/espeak-ng/assets/11532015/2fed6ce5-dc59-46e3-8b40-dac0bf54801d)

**Comment by Likivik** [comment]
> **there is no voice installed.**
> 
> ![image](https://private-user-images.githubusercontent.com/11532015/328061194-2fed6ce5-dc59-46e3-8b40-dac0bf54801d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE0OTg4ODMsIm5iZiI6MTcyMTQ5ODU4MywicGF0aCI6Ii8xMTUzMjAxNS8zMjgwNjExOTQtMmZlZDZjZTUtZGM1OS00NmUzLThiNDAtZGFjMGJmNTQ4MDFkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzIwVDE4MDMwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ2NzE4NTFkNTE5OWE0NjU5NDg1MjgyMDkyNDAwMGVjY2I3ZDUyOTA3ZDUwZmIyMmFkODgzYTVlMTJlMDM0NzcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.bXNyvDvubHpKRqeRzMM1a_sEN3Sn3D-quod5orwSTsc)

For now, please follow the instructions from that answer up top: https://github.com/espeak-ng/espeak-ng/issues/1078#issuecomment-1173619939

**Comment by DiagonalArg** [comment]
> for now, I can suggest this workaround:
> 
>     1. install espeak
>     2. check that app displays that 0 voices installed
>     3. select espeak as the tts in android settings
>     4. somehow send text to espeak
>     5. now app should display that 131 or so voices is available
>     6. stop espeak app or reboot the phone
> 
> now espeak should work.

Not working for me on grapheneos, despite reboot.  I'm using your release from [here](https://github.com/espeak-ng/espeak-ng/issues/1819#issuecomment-2323360297).

**Comment by alex19EP** [comment]
@DiagonalArg hello. Does the application show that voices are installed?

**Comment by alex19EP** [comment]
hmm. do you have another tts on the system? can you try to switch it back and forth several times?

we actually need to somehow figure out a way to separate the installation of voice data from the initialization of tts, or finally understand why sometimes this installation does not happen.

**Comment by DiagonalArg** [comment]
> hmm. do you have another tts on the system? can you try to switch it back and forth several times?

I don't.

> we actually need to somehow figure out a way to separate the installation of voice data from the initialization of tts, or finally understand why sometimes this installation does not happen.

I'm just noticing that eSpeak-ng doesn't seem to request any permissions.  Notifications?  File access?  When I try to import a dictionary, it doesn't see any filesystem directories.

**Comment by beqabeqa473** [comment]
Espeak-ng itself has this problem. it is not reinitializing when data is extracted. I think this is very old bug.

**Comment by DiagonalArg** [comment]
> hmm. do you have another tts on the system? can you try to switch it back and forth several times?

If there's another Free-as-in-Freedom TTS system available, I can install it on my GrapheneOS phone; but as far as I could find, your project is it!

**Comment by alex19EP** [comment]
you can try [RHVoice](https://github.com/RHVoice/RHVoice)

**Comment by alex19EP** [comment]
you can try [RHVoice](https://github.com/RHVoice/RHVoice)

**Comment by DiagonalArg** [comment]
Ok, I installed RHVoice and was now able to toggle between both it and espeak.

**Comment by hms0411rt** [comment]
@alex19EP as hotfix, I would suggest adding a note somewhere (in the readme in the PlayStore/F-Droid Store or in the app itself) about the workaround.
